dnf5-plugins/reposync_plugin/reposync.cpp

```diff
@@ -0,0 +1,420 @@
+    if (!metadata_path.empty()) {
+        repo_path = std::filesystem::absolute(metadata_path);
+        if (!norepopath_option->get_value()) {
+            repo_path /= repo.get_id();
+        }
+        // resolve '.', '..', and existing symlinks in the repo_path
+        repo_path = std::filesystem::weakly_canonical(repo_path);
+    } else {
+        repo_path = repo_download_path(repo);
+    }
+    auto & optional_metadata_option = get_context().get_base().get_config().get_optional_metadata_types_option();
+    if (!optional_metadata_option.get_value().contains(libdnf5::METADATA_TYPE_ALL)) {
+        optional_metadata_option.set(libdnf5::METADATA_TYPE_ALL);
+    }
+    repo.download_metadata(repo_path);
+}
+
+void ReposyncCommand::run() {
+    auto & context = get_context();
+    libdnf5::repo::RepoQuery repos_query(context.get_base());
+    repos_query.filter_enabled(true);
+    std::vector<std::string_view> schemes{"https://", "file://", "http://", "ftp://"};
+    for (const auto & repo : repos_query) {
+        const auto pkg_list = get_packages_list(*repo);
+        if (urls_option->get_value()) {
+            if (download_metadata_option->get_value()) {
+                // get list of repository remote locations (mirrors + base_url)
+                std::vector<std::string> remote_locations;
+                for (const auto & mirror : repo->get_mirrors()) {
+                    remote_locations.emplace_back(mirror);
+                }
+                for (const auto & base_url : repo->get_config().get_baseurl_option().get_value()) {
+                    remote_locations.emplace_back(base_url);
+                }
+                // find first available mirror prefering file and https schemes
+                std::string repo_location{};
+                for (const auto & scheme : schemes) {
+                    for (const auto & mirror : remote_locations) {
+                        if (mirror.starts_with(scheme)) {
+                            repo_location = mirror;
+                            break;
+                        }
+                    }
+                    if (!repo_location.empty()) {
+                        break;
+                    }
+                }
+                if (repo_location.empty()) {
+                    std::cerr << libdnf5::utils::sformat(_("Failed to get mirror for metadata.")) << std::endl;
+                    continue;
+                }
+                for (const auto & [md_type, md_location] : repo->get_metadata_locations()) {
+                    std::cout << join_url(repo_location, md_location) << std::endl;
+                }
+            }
+            for (const auto & [pth, pkg] : pkg_list) {
+                auto urls = pkg.get_remote_locations();
+                if (urls.empty()) {
+                    std::cerr << libdnf5::utils::sformat(_("Failed to get mirror for package: \"{}\""), pkg.get_name())
+                              << std::endl;
+                    continue;
```

Comment:
The `continue` doesn't seem to be needed?

Reply:
Removed.